test/markdownlint-cli2-test-cases.js

```diff
@@ -629,15 +641,19 @@ const testCases =
       "invalid.markdownlint-cli2.jsonc",
       /(?:Unexpected end)|(?:Expected property name)/u
     ],
-    [ "invalid.markdownlint-cli2.cjs", /Unexpected end of input/u ],
-    [ "invalid.markdownlint-cli2.mjs", /Unexpected end of input/u ],
+    // eslint-disable-next-line max-len
+    [ "invalid.markdownlint-cli2.cjs", /Unable to require or import module '\.\/invalid.markdownlint-cli2.cjs/u ],
```

Comment:
I don't understand why this set of changes is necessary. Tests pass in Actions without this change. Are you on a version of Node other than 16/18/20? Can you tell what's different for you?

Reply:
> Tests pass in Actions without this change.

[Tests are passing](https://github.com/DavidAnson/markdownlint-cli2/actions/runs/5865467884/job/15929413139?pr=191#step:5:404) *with* these changes as well. :confused: 

> Can you tell what's different for you?

I was on Node v16.13.0(LTS) where the tests are failing without the changes. Now I've switched to v16.20.0(LTS) and the tests are passing *with and without* the changes. So I've reverted the changes.

***
Recently you implemented the AggregateError:

<details>
<summary>Details</summary>

```log
`AggregateError: Unable to require or import module './invalid.markdownlint-cli2.cjs'.␊
      at importOrRequireResolve (/home/koder/GitHub/markdownlint-cli2/markdownlint-cli2.js:91:11)␊
      at async readOptionsOrConfig (/home/koder/GitHub/markdownlint-cli2/markdownlint-cli2.js:155:15)␊
      at async main (/home/koder/GitHub/markdownlint-cli2/markdownlint-cli2.js:890:7)␊
      at async /home/koder/GitHub/markdownlint-cli2/markdownlint-cli2.js:984:26 {␊
    [errors]: [␊
      /home/koder/GitHub/markdownlint-cli2/test/config-files/cfg/invalid.markdownlint-cli2.cjs:6␊
      ␊
      ␊
      ␊
      SyntaxError: Unexpected end of input␊
          at Object.compileFunction (node:vm:360:18)␊
          at wrapSafe (node:internal/modules/cjs/loader:1124:15)␊
          at Module._compile (node:internal/modules/cjs/loader:1160:27)␊
          at Object.Module._extensions..js (node:internal/modules/cjs/loader:1250:10)␊
          at Module.load (node:internal/modules/cjs/loader:1074:32)␊
          at Function.Module._load (node:internal/modules/cjs/loader:909:12)␊
          at Module.require (node:internal/modules/cjs/loader:1098:19)␊
          at require (node:internal/modules/cjs/helpers:108:18)␊
          at resolveAndRequire (/home/koder/GitHub/markdownlint-cli2/resolve-and-require.js:16:10)␊
          at importOrRequireResolve (/home/koder/GitHub/markdownlint-cli2/markdownlint-cli2.js:77:14),␊
      /home/koder/GitHub/markdownlint-cli2/test/config-files/cfg/invalid.markdownlint-cli2.cjs:6␊
      ␊
      ␊
      ␊
      SyntaxError: Unexpected end of input␊
          at Object.compileFunction (node:vm:360:18)␊
          at wrapSafe (node:internal/modules/cjs/loader:1124:15)␊
          at Module._compile (node:internal/modules/cjs/loader:1160:27)␊
          at Object.Module._extensions..js (node:internal/modules/cjs/loader:1250:10)␊
          at Module.load (node:internal/modules/cjs/loader:1074:32)␊
          at Function.Module._load (node:internal/modules/cjs/loader:909:12)␊
          at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)␊
          at ModuleJob.run (node:internal/modules/esm/module_job:193:25)␊
          at async Promise.all (index 0)␊
          at async ESMLoader.import (node:internal/modules/esm/loader:530:24)␊
    ]␊
  }␊
  `
```

</details>

But these tests were not updated at that time. I think it's better to update them than to keep them as it is.

You could [add .nvmrc](https://github.com/nvm-sh/nvm#nvmrc) to lock the versions required for the project. This will help contributors using `nvm` or `fnm`.

Reply:
Thanks for figuring that out! I made a note to update the RegExp after I'm done with some other test changes I'm in the middle of. I prefer to keep them out of this commit to avoid combining unrelated work.